### ISSUE TITLE
Expose Nested Dictionary Fields in the Query builder

### ISSUE THREAD

**Issue by Addvilz** [metabase/metabase#1328]
Any chance we could query against nested fields in Mongo? Example document:

``` json
{
    "id": 1234,
    "properties": {
        "key": "val",
        "otherkey": 123
    },
    "value": 33,
    "amount": 12
}
```

In this case we should really use ability to access `properties.*` as any other field.

**Comment by salsakran** [comment]
We've been sitting on this for ages. What needs to happen for this to work on the front end?

**Comment by aelanic** [comment]
Raised the same issue in #2807 . Any idea when is this getting fixed?

**Comment by camsaul** [comment]
It sounds like this may be fixed already, since we have #3173 open to fix behaviors with this. @salsakran do you know if that's the case?

**Comment by salsakran** [comment]
not sure. Let me poke around our mongo dataset.

**Comment by gierschv** [comment]
Hey there,

No, it's not fixed, I have the nested fields represented as JSON in tables:

![screenshot 2016-10-28 22 23 30](https://cloud.githubusercontent.com/assets/396537/19821161/3bbae85c-9d5d-11e6-9e92-c09d8e71e85b.png)

They are displayed in a single level (without the parent fields) in the _filtered by_ selector, and in the visualization settings only the root fields are listed.

Vincent

**Comment by camsaul** [comment]
@gierschv this issue isn't taking about how nested fields are displayed but rather if you can see them as options in things like the filter menu. e.g. if you're able to do something like filter by `statistics.sessions == 2`

**Comment by Addvilz** [comment]
Afaik sub document keys are currently merged together with parent keys. Not okay imho.

**Comment by tlrobinson** [comment]
To be clear, you _can_ currently filter, group, aggregate, and sort nested fields, the one remaining piece is returning the nested fields as individual columns in "raw data" queries.

@camsaul is there a reason we're not doing that by default?

FYI manually adding the nested fields using the "additional fields" branch does work (but that branch is a prototype) #3043

**Comment by camsaul** [comment]
I think the core issues here have been addressed by #3738.

As far as automatically flattening all nested keys for bare rows query, that's something we could potentially do, but it's more of a new feature with a different scope than this issue. I've opened #4098 for that idea so we can close this one out.
